pkg/webhooks/kafkatopic_validator.go

```diff
@@ -154,37 +162,62 @@ func (s *KafkaTopicValidator) checkKafka(ctx context.Context, topic *banzaicloud
 		return nil, errors.WrapIff(err, fmt.Sprintf("failed to list topics for kafka cluster: %s", topic.Spec.ClusterRef.Name))
 	}
 
+	var allErrs field.ErrorList
 	// The topic exists
 	if existing != nil {
 		// Check if this is the correct CR for this topic
 		topicCR := &banzaicloudv1alpha1.KafkaTopic{}
 		if err := s.Client.Get(ctx, types.NamespacedName{Name: topic.Name, Namespace: topic.Namespace}, topicCR); err != nil {
+			// Checking that the validation request is update
 			if apierrors.IsNotFound(err) {
-				// User is trying to overwrite an existing topic - bad user
-				logMsg := fmt.Sprintf("topic already exists on kafka cluster '%s'", topic.Spec.ClusterRef.Name)
-				return field.Invalid(field.NewPath("spec").Child("name"), topic.Spec.Name, logMsg), nil
+				if manager, ok := topic.GetAnnotations()[TopicManagedByAnnotationKey]; !ok || strings.ToLower(manager) != TopicManagedByKoperatorAnnotationValue {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("name"), topic.Spec.Name,
+						fmt.Sprintf(`topic "%s" already exists on kafka cluster and it is not managed by Koperator,
+					if you want it to be managed by Koperator making you able to modify its configuration through a KafkaTopic CR,
+					add this "%s: %s" annotation to this KafkaTopic CR`, topic.Spec.Name, TopicManagedByAnnotationKey, TopicManagedByKoperatorAnnotationValue)))
+				}
+				// Comparing KafkaTopic configuration with the existing
+				if existing.NumPartitions != topic.Spec.Partitions {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("partitions"), topic.Spec.Partitions,
+						fmt.Sprintf(`initial KafkaTopic partition number must be the same as what the existing kafka topic has  (given: %v present: %v)`, topic.Spec.Partitions, existing.NumPartitions)))
```

Comment:
I am a bit confused about this "initial" wording -- it is leading me to the partition number that the users specified when the kafka topic was initially created. But we are actually comparing the new partition number against the existing value

Same confusion while reading the other "initial" messages in this function

Reply:
I have changed it.  What you think? Is it clear now?

Reply:
Perhaps having the following is clear enough:

`"When creating KafkaTopic CR for existing topic, its partition number must be the same as what the existing kafka topic has..."`

What do you think?